renderdoc/driver/gl/gl_hooks_vrapi.cpp

```diff
@@ -0,0 +1,288 @@
+extern "C" {
+
+__attribute__((visibility("default"))) ovrMobile *vrapi_EnterVrMode(const ovrModeParms *parms)
+{
+  if(vrapi_hooks.vrapi_EnterVrMode_real == NULL)
+  {
+    vrapi_hooks.SetupHooks();
+  }
+
+  return vrapi_hooks.vrapi_EnterVrMode_real(parms);
+}
+
+__attribute__((visibility("default"))) void vrapi_LeaveVrMode(ovrMobile *ovr)
+{
+  if(vrapi_hooks.vrapi_LeaveVrMode_real == NULL)
+  {
+    vrapi_hooks.SetupHooks();
+  }
+
+  vrapi_hooks.vrapi_LeaveVrMode_real(ovr);
+}
+
+__attribute__((visibility("default"))) ovrTextureSwapChain *vrapi_CreateTextureSwapChain2(
+    ovrTextureType type, ovrTextureFormat format, int width, int height, int levels, int bufferCount)
+{
+  if(vrapi_hooks.vrapi_CreateTextureSwapChain2_real == NULL ||
+     vrapi_hooks.vrapi_GetTextureSwapChainHandle_real == NULL ||
+     vrapi_hooks.vrapi_GetTextureSwapChainLength_real == NULL)
+  {
+    vrapi_hooks.SetupHooks();
+  }
+
+  ovrTextureSwapChain *texture_swapchain = vrapi_hooks.vrapi_CreateTextureSwapChain2_real(
+      type, format, width, height, levels, bufferCount);
+
+  int tex_count = vrapi_hooks.vrapi_GetTextureSwapChainLength_real(texture_swapchain);
+
+  if(m_GLDriver)
+  {
+    for(int i = 0; i < tex_count; ++i)
+    {
+      GLuint tex = vrapi_hooks.vrapi_GetTextureSwapChainHandle_real(texture_swapchain, i);
+      GLenum internalformat = GetInternalFormat(format);
+      GLenum textureType = GetTextureType(type);
+
+      m_GLDriver->CreateVRAPITextureSwapChain(tex, textureType, internalformat, width, height);
+    }
+  }
+
+  return texture_swapchain;
+}
+
+__attribute__((visibility("default"))) ovrTextureSwapChain *vrapi_CreateTextureSwapChain(
+    ovrTextureType type, ovrTextureFormat format, int width, int height, int levels, bool buffered)
+{
+  if(vrapi_hooks.vrapi_CreateTextureSwapChain_real == NULL ||
+     vrapi_hooks.vrapi_GetTextureSwapChainHandle_real == NULL ||
+     vrapi_hooks.vrapi_GetTextureSwapChainLength_real == NULL)
+  {
+    vrapi_hooks.SetupHooks();
+  }
+
+  ovrTextureSwapChain *texture_swapchain =
+      vrapi_hooks.vrapi_CreateTextureSwapChain_real(type, format, width, height, levels, buffered);
+
+  int tex_count = vrapi_hooks.vrapi_GetTextureSwapChainLength_real(texture_swapchain);
+
+  for(int i = 0; i < tex_count; ++i)
+  {
+    GLuint tex = vrapi_hooks.vrapi_GetTextureSwapChainHandle_real(texture_swapchain, i);
+    GLenum internalformat = GetInternalFormat(format);
+    GLenum textureType = GetTextureType(type);
+
+    m_GLDriver->CreateVRAPITextureSwapChain(tex, textureType, internalformat, width, height);
+  }
+
+  return texture_swapchain;
+}
+
+__attribute__((visibility("default"))) void vrapi_SubmitFrame(ovrMobile *ovr,
+                                                              const ovrFrameParms *parms)
+{
+  if(vrapi_hooks.vrapi_SubmitFrame_real == NULL || vrapi_hooks.vrapi_GetSystemPropertyInt_real == NULL)
+  {
+    vrapi_hooks.SetupHooks();
+  }
+
+  if(m_GLDriver)
+  {
+    SCOPED_LOCK(glLock);
```

Comment:
Well remembered about the GL lock! I'd forgotten that this was needed in case of multi-threading code using vrapi which will probably be common.

I think this will be needed in the other functions that call into `m_GLDriver` right? `vrapi_CreateTextureSwapChain` and `vrapi_CreateTextureSwapChain2` ?